Apps/phipqualityscore/ipqualityscore_connector.py

```diff
@@ -0,0 +1,397 @@
+            self.debug_print('Response from server is not a valid JSON {}'.format(err))
+            self.save_progress('Response from server is not a valid JSON')
+            self.save_progress(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST)
+            return self.set_status(phantom.APP_ERROR)
+
+        if 'success' in result and result['success'] is True:
+            self.save_progress(IPQUALITYSCORE_SUCC_CONNECTIVITY_TEST)
+            return self.set_status(phantom.APP_SUCCESS)
+
+        self.save_progress(IPQUALITYSCORE_ERR_CONNECTIVITY_TEST)
+        return self.set_status(phantom.APP_ERROR)
+
+    def create_req_url(self, urltype, param, app_key):
+        if urltype == "url":
+            req_url = IPQUALITYSCORE_API_URL_CHECKER.format(
+                apikey=app_key, url=urllib.parse.quote_plus(param['url']))
+        elif urltype == "ip":
+            req_url = IPQUALITYSCORE_API_IP_REPUTATION.format(
+                apikey=app_key, ip=(param['ip']))
+        elif urltype == "email":
+            req_url = IPQUALITYSCORE_API_EMAIL_VALIDATION.format(
+                apikey=app_key, email=param['email'])
+        else:
+            req_url = ''
+        # optional parameters
+        optional_param = ''
+        if param.get('strictness') is not None:
+            optional_param = "{}&strictness={}".format(optional_param, param.get('strictness'))
+        if param.get('user_agent') is not None:
+            optional_param = "{}&user_agent={}".format(optional_param, param.get('user_agent'))
+        if param.get('user_language') is not None:
+            optional_param = "{}&user_language={}".format(optional_param, param.get('user_language'))
+        if param.get('fast') is not None:
+            optional_param = "{}&fast={}".format(optional_param, param.get('fast'))
+        if param.get('mobile') is not None:
+            optional_param = "{}&mobile={}".format(optional_param, param.get('mobile'))
+        if param.get('allow_public_access_points') is not None:
+            optional_param = "{}&allow_public_access_points={}".format(optional_param, param.get('allow_public_access_points'))
+        if param.get('lighter_penalties') is not None:
+            optional_param = "{}&lighter_penalties={}".format(optional_param, param.get('lighter_penalties'))
+        if param.get('transaction_strictness') is not None:
+            optional_param = "{}&transaction_strictness={}".format(optional_param, param.get('transaction_strictness'))
+        if param.get('timeout') is not None:
+            optional_param = "{}&timeout={}".format(optional_param, param.get('timeout'))
+        if param.get('suggest_domain') is not None:
+            optional_param = "{}&suggest_domain={}".format(optional_param, param.get('suggest_domain'))
+        if param.get('abuse_strictness') is not None:
+            optional_param = "{}&abuse_strictness={}".format(optional_param, param.get('abuse_strictness'))
```

Comment:
for readability can we aggregate these in a dictionary and join it as a string in the end?

for example:
```
optional_params = {
    'strictness': param.get('strictness'),
    'user_agent': param.get('user_agent'),
    ...
}
query_string = '&'.join(f'{k}={v}' for k, v in optional_params if v)
```

Reply:
Fixed